### PR TITLE
Remove recompose

### PR DESCRIPTION
**Pull request by emmenko**
- drops `recompose` dependency
- implements `get-display-name` utility function in each package that needs it
- clean up `fetch{User,Project}` components and rewrite tests using RTL
- rewrite `branchOnPermissions` component and add test using RTL

### CODE REVIEW

**Comment by emmenko** [comment]
@jonnybel there are some changes related to the login/logout components, make sure to migrate them as well to the new app.

File: packages/application-components/package.json [hunk 1]
```diff
@@ -66,6 +66,6 @@
     "react": "16.x",
     "react-dom": "16.x",
     "react-intl": "2.x",
-    "react-testing-library": ">=5 <7"
+    "react-testing-library": "6.x || 7.x"
```
Comment (montezume) [inline packages/application-components/package.json]: oh nice, I didn't know this syntax existed

File: packages/application-shell/src/components/login-sso-callback/login-sso-callback.js [hunk 2]
```diff
@@ -32,14 +31,15 @@ export class LoginSSOCallback extends React.PureComponent {
...
     }).isRequired,
-    redirectTo: PropTypes.func.isRequired,
     requestAccessToken: PropTypes.func.isRequired,
   };
 
   state = {
     hasAuthenticationFailed: false,
   };
 
+  redirectTo = targetUrl => window.location.replace(targetUrl);
```
Comment (montezume) [inline packages/application-shell/src/components/login-sso-callback/login-sso-callback.js]: this is failing the tests, is there any reason why you did this?

Comment (montezume) [reply packages/application-shell/src/components/login-sso-callback/login-sso-callback.js]: jsdom doesn't implement window.location.replace

Comment (emmenko) [reply packages/application-shell/src/components/login-sso-callback/login-sso-callback.js]: Ah I’ll fix that no worries

**Comment by montezume** [comment]
Okay I pushed some commits to fix the failing tests 🎉

**Comment by emmenko** [comment]
Ah thanks! 🙌

**Pull request merged**